Unable to create class 'org.eclipse.chemclipse.ux.extension.xxd.ui.parts.MoleculePart'

```
!ENTRY org.eclipse.e4.ui.workbench 4 0 2021-08-10 15:08:14.410
!MESSAGE Unable to create class 'org.eclipse.chemclipse.ux.extension.xxd.ui.parts.MoleculePart' from bundle '285'
!STACK 0
org.eclipse.e4.core.di.InjectionException: java.lang.NullPointerException
	at org.eclipse.e4.core.internal.di.ConstructorRequestor.execute(ConstructorRequestor.java:52)
	at org.eclipse.e4.core.internal.di.InjectorImpl.internalMake(InjectorImpl.java:410)
	at org.eclipse.e4.core.internal.di.InjectorImpl.make(InjectorImpl.java:331)
	at org.eclipse.e4.core.contexts.ContextInjectionFactory.make(ContextInjectionFactory.java:190)
	at org.eclipse.e4.ui.internal.workbench.ReflectionContributionFactory.createFromBundle(ReflectionContributionFactory.java:105)
	at org.eclipse.e4.ui.internal.workbench.ReflectionContributionFactory.doCreate(ReflectionContributionFactory.java:74)
	at org.eclipse.e4.ui.internal.workbench.ReflectionContributionFactory.create(ReflectionContributionFactory.java:56)
	at org.eclipse.e4.ui.workbench.renderers.swt.ContributedPartRenderer.createWidget(ContributedPartRenderer.java:129)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.createWidget(PartRenderingEngine.java:1012)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeCreateGui(PartRenderingEngine.java:672)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeCreateGui(PartRenderingEngine.java:778)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.access$0(PartRenderingEngine.java:749)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$2.run(PartRenderingEngine.java:743)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.createGui(PartRenderingEngine.java:727)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.subscribeTopicToBeRendered(PartRenderingEngine.java:157)
	at sun.reflect.GeneratedMethodAccessor51.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:55)
	at org.eclipse.e4.core.di.internal.extensions.EventObjectSupplier$DIEventHandler.handleEvent(EventObjectSupplier.java:88)
	at org.eclipse.equinox.internal.event.EventHandlerWrapper.handleEvent(EventHandlerWrapper.java:201)
	at org.eclipse.equinox.internal.event.EventHandlerTracker.dispatchEvent(EventHandlerTracker.java:196)
	at org.eclipse.equinox.internal.event.EventHandlerTracker.dispatchEvent(EventHandlerTracker.java:1)
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230)
	at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:148)
	at org.eclipse.equinox.internal.event.EventAdminImpl.dispatchEvent(EventAdminImpl.java:135)
	at org.eclipse.equinox.internal.event.EventAdminImpl.sendEvent(EventAdminImpl.java:78)
	at org.eclipse.equinox.internal.event.EventComponent.sendEvent(EventComponent.java:39)
	at org.eclipse.e4.ui.services.internal.events.EventBroker.send(EventBroker.java:52)
	at org.eclipse.e4.ui.internal.workbench.UIEventPublisher.notifyChanged(UIEventPublisher.java:60)
	at org.eclipse.emf.common.notify.impl.BasicNotifierImpl.eNotify(BasicNotifierImpl.java:374)
	at org.eclipse.e4.ui.model.application.ui.impl.UIElementImpl.setToBeRendered(UIElementImpl.java:304)
	at org.eclipse.e4.ui.internal.workbench.ModelServiceImpl.showElementInWindow(ModelServiceImpl.java:629)
	at org.eclipse.e4.ui.internal.workbench.ModelServiceImpl.bringToTop(ModelServiceImpl.java:598)
	at org.eclipse.e4.ui.internal.workbench.PartServiceImpl.delegateBringToTop(PartServiceImpl.java:788)
	at org.eclipse.e4.ui.internal.workbench.PartServiceImpl.activate(PartServiceImpl.java:759)
	at org.eclipse.e4.ui.internal.workbench.PartServiceImpl.activate(PartServiceImpl.java:681)
	at org.eclipse.e4.ui.internal.workbench.PartServiceImpl.activate(PartServiceImpl.java:676)
	at org.eclipse.e4.ui.internal.workbench.PartServiceImpl.showPart(PartServiceImpl.java:1229)
	at org.eclipse.e4.ui.internal.workbench.ApplicationPartServiceImpl.showPart(ApplicationPartServiceImpl.java:140)
	at org.eclipse.chemclipse.ux.extension.ui.support.PartSupport.showPart(PartSupport.java:288)
	at org.eclipse.chemclipse.ux.extension.ui.support.PartSupport.setPartVisibility(PartSupport.java:368)
	at org.eclipse.chemclipse.ux.extension.ui.support.PartSupport.togglePartVisibility(PartSupport.java:235)
	at org.eclipse.chemclipse.ux.extension.ui.support.PartSupport.togglePartVisibility(PartSupport.java:429)
	at org.eclipse.chemclipse.ux.extension.xxd.ui.toolbar.AbstractPartHandler.action(AbstractPartHandler.java:119)
	at org.eclipse.chemclipse.ux.extension.xxd.ui.toolbar.AbstractGroupHandler.setPartStatus(AbstractGroupHandler.java:147)
	at org.eclipse.chemclipse.ux.extension.xxd.ui.toolbar.AbstractGroupHandler.execute(AbstractGroupHandler.java:47)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:55)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:318)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:324)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:252)
	at org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:161)
	at org.eclipse.e4.ui.workbench.renderers.swt.DirectContributionItem.executeItem(DirectContributionItem.java:130)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.handleWidgetSelection(AbstractContributionItem.java:446)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.lambda$2(AbstractContributionItem.java:472)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:86)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:5686)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1370)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4940)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4518)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1170)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1059)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:153)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:667)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:597)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:148)
	at org.eclipse.chemclipse.rcp.app.ui.internal.support.ApplicationSupportDefault.start(ApplicationSupportDefault.java:26)
	at org.eclipse.chemclipse.rcp.app.ui.Application.start(Application.java:61)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:656)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:592)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1498)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1471)
Caused by: java.lang.NullPointerException
	at org.eclipse.chemclipse.ux.extension.xxd.ui.swt.ExtendedMoleculeUI.createSettingsButton(ExtendedMoleculeUI.java:455)
	at org.eclipse.chemclipse.ux.extension.xxd.ui.swt.ExtendedMoleculeUI.createToolbarMain(ExtendedMoleculeUI.java:167)
	at org.eclipse.chemclipse.ux.extension.xxd.ui.swt.ExtendedMoleculeUI.createControl(ExtendedMoleculeUI.java:123)
	at org.eclipse.chemclipse.ux.extension.xxd.ui.swt.ExtendedMoleculeUI.<init>(ExtendedMoleculeUI.java:101)
	at org.eclipse.chemclipse.ux.extension.xxd.ui.parts.MoleculePart.createControl(MoleculePart.java:39)
	at org.eclipse.chemclipse.ux.extension.xxd.ui.parts.MoleculePart.createControl(MoleculePart.java:1)
	at org.eclipse.chemclipse.ux.extension.xxd.ui.parts.AbstractPart.<init>(AbstractPart.java:27)
	at org.eclipse.chemclipse.ux.extension.xxd.ui.parts.MoleculePart.<init>(MoleculePart.java:33)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.eclipse.e4.core.internal.di.ConstructorRequestor.execute(ConstructorRequestor.java:40)
	... 87 more
```

--- a/chemclipse/plugins/org.eclipse.chemclipse.ux.extension.xxd.ui/src/org/eclipse/chemclipse/ux/extension/xxd/ui/swt/ExtendedMoleculeUI.java
+++ b/chemclipse/plugins/org.eclipse.chemclipse.ux.extension.xxd.ui/src/org/eclipse/chemclipse/ux/extension/xxd/ui/swt/ExtendedMoleculeUI.java
@@ -136,9 +136,11 @@ public class ExtendedMoleculeUI extends Composite implements IExtendedPartUI {
 		 * Services
 		 */
 		Object[] moleculeImageServices = Activator.getDefault().getMoleculeImageServices();
-		comboViewerServices.setInput(moleculeImageServices);
-		if(moleculeImageServices.length >= 1) {
-			comboViewerServices.getCombo().select(0);
+		if(moleculeImageServices != null) {
+			comboViewerServices.setInput(moleculeImageServices);
+			if(moleculeImageServices.length >= 1) {
+				comboViewerServices.getCombo().select(0);
+			}
 		}
 		/*
 		 * Input Types
@@ -452,12 +454,14 @@ public class ExtendedMoleculeUI extends Composite implements IExtendedPartUI {
 		 * Additional pages.
 		 */
 		Object[] moleculeImageServices = Activator.getDefault().getMoleculeImageServices();
-		for(Object object : moleculeImageServices) {
-			if(object instanceof IMoleculeImageService) {
-				IMoleculeImageService moleculeImageService = (IMoleculeImageService)object;
-				Class<? extends IWorkbenchPreferencePage> preferencePage = moleculeImageService.getPreferencePage();
-				if(preferencePage != null) {
-					preferencePages.add(preferencePage);
+		if(moleculeImageServices != null) {
+			for(Object object : moleculeImageServices) {
+				if(object instanceof IMoleculeImageService) {
+					IMoleculeImageService moleculeImageService = (IMoleculeImageService)object;
+					Class<? extends IWorkbenchPreferencePage> preferencePage = moleculeImageService.getPreferencePage();
+					if(preferencePage != null) {
+						preferencePages.add(preferencePage);
+					}
 				}
 			}
 		}

